### PR TITLE
Support for DataBreakpointAccessType condition

### LINKED ISSUE #113922: Support for editing 'accessType' property of data breakpoints
Regarding #83649 ( @isidorn  )

As discussed in previous issue about DataBreakpoints support, it can be great if VS Code can surface a user interface to create and edit Data Breakpoints regarding to their attributes, such as `read`, `write`, `readWrite` for access type and `condition`/`hitCondition`.

Thanks

### PR DESCRIPTION
**Pull request by yannickowow**
This PR fixes #113922

Hi, 
Here you will find a draft PR to support DataBreakpoint access type.
When right-clicking on a DataBreakpoint variable, you can by this way select an action based on `DataBreakpointAccessTypes[]` response.
By default, an empty array assumes `Break When Value Changes` (it was the current behaviour).

Here you will find a record about this new behaviour:

https://user-images.githubusercontent.com/45362067/109398556-e5c83000-793d-11eb-83b5-58176c388e9b.mp4

For a better experience, I would suggest here a radiobutton instead of checkbox... Any ideas ?

I did update breakpointTests as well to ensure `accessType` is correctly set.

Thanks in advance !

### CODE REVIEW

**Comment by isidorn** [comment]
@yannickowow thanks for your PR and sorry for the slow response.
I glanced over the PR and it looks good overall. First I would like to ask you do the following
1. Can you please merge latest from `main` and resolve conflicts
2. The `updateDataBreakpoint` work can be split up from the inital support of all `accessTypes`. Due to that it would be great if in this PR you could only leave the support for all `accessTypes` and all the `updateDataBreakpoint` work to be put in another PR. Once we merge this PR and once we gain enough user feedback we can consider merging in the other `updateDataBreakpoint` one

I hope my suggestion make sense and I wish you a nice day

**Comment by yannickowow** [comment]
If I understood correctly, you are considering actions to support `accessTypes` but not updating them in `Breakpoints` view (used by `updateDataBreakpoint`) ?
If so, it is done. 👍 
I did remove actions associated to DataBreakpoints in the Breakpoints view.

Regards.

File: src/vs/workbench/contrib/debug/common/debug.ts
Comment (isidorn) [inline src/vs/workbench/contrib/debug/common/debug.ts:61]: "item type" should be "access type"

File: src/vs/workbench/contrib/debug/browser/debug.contribution.ts
Comment (isidorn) [inline src/vs/workbench/contrib/debug/browser/debug.contribution.ts:142]: These commands should not all have the same `order` of 200. I suggest to assign them different order based on the order you think they should appear

File: src/vs/workbench/contrib/debug/browser/breakpointsView.ts
Comment (isidorn) [inline src/vs/workbench/contrib/debug/browser/breakpointsView.ts:107]: I do not understand why are we setting these context keys in the breakpoints view, since we are only using them in the `variablesView` for command enablment. Can all of this work in the breakpoints view be removed, or am I missing some connection?

**Review by isidorn — CHANGES_REQUESTED** [review]

**Comment by isidorn** [comment]
Great work, thanks again for your PR. I have left some comments in the code, once you tackle that let me know and I can do another review.

Since we will probably merge this in this milestone it might be cool to write a test plan item so we cover this. Here's a good template https://github.com/microsoft/vscode/issues/79799 @yannickowow if you have time maybe you could write a test plan item and ping me @isidorn on it
Though it might be cool if mock-debug supported these new changes so we could write a test plan item using mock debug @weinand

File: src/vs/workbench/contrib/debug/common/debug.ts
Comment (yannickowow) [reply src/vs/workbench/contrib/debug/common/debug.ts:61]: Correct

File: src/vs/workbench/contrib/debug/browser/breakpointsView.ts
Comment (yannickowow) [reply src/vs/workbench/contrib/debug/browser/breakpointsView.ts]: You are correct, it was previously used to update breakpoints in the `Breakpoints` view, as checkboxes. Since I removed previous code about it, I must remove this part as well.

**Comment by isidorn** [comment]
Fantastic work. Merging in, thanks a lot for your contribution ☀️

**Pull request merged**
